lib/ansible/plugins/test/core.py

```diff
@@ -114,6 +115,7 @@ def regex(value='', pattern='', ignorecase=False, multiline=False, match_type='s
         This is likely only useful for `search` and `match` which already
         have their own filters.
     '''
+    value = to_text(value, errors='surrogate_or_strict')
```

Comment:
I'd probably add a comment about why this is done (the side-effect) since it's not obvious just from the code.